### ISSUE TITLE
«imagecolorsforindex(): Argument #2 ($color) is out of range» for some GIF images

### ISSUE THREAD

**Issue by dmitrii-fediuk** [27estore/site#67]
```
{
    "mage2pro/core": "11.4.2",
    "Magento": "2.4.6-p4",
    "PHP": "8.2.16",
    "URL": "https://www.27estore.com/one1-aluminum-frame-white-door",
    "Time": "2024-10-06 11:53:15",
    "Referer": "",
    "IP Address": "3.144.106.73",
    "Request Method": "GET",
    "User-Agent": "Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; ClaudeBot/1.0; +claudebot@anthropic.com)"
}
EXCEPTION
imagecolorsforindex(): Argument #2 ($color) is out of range
************************************
[]
************************************
1	imagecolorsforindex
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Image/Adapter/Gd2.php#L351
	vendor/magento/framework/Image/Adapter/Gd2.php:351

2	Magento\Framework\Image\Adapter\Gd2::applyTransparency
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Image/Adapter/Gd2.php#L351
	vendor/magento/framework/Image/Adapter/Gd2.php:351

3	Magento\Framework\Image\Adapter\Gd2::fillBackgroundColor
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Image/Adapter/Gd2.php#L292
	vendor/magento/framework/Image/Adapter/Gd2.php:292

4	Magento\Framework\Image\Adapter\Gd2::resize
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Image/Adapter/Gd2.php#L436
	vendor/magento/framework/Image/Adapter/Gd2.php:436

5	Magento\Framework\Image::resize
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Image.php#L115
	vendor/magento/framework/Image.php:115

6	Magento\Catalog\Model\Product\Image::resize
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Model/Product/Image.php#L520
	vendor/magento/module-catalog/Model/Product/Image.php:520

7	Magento\Catalog\Helper\Image::applyScheduledActions
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Helper/Image.php#L479
	vendor/magento/module-catalog/Helper/Image.php:479

8	Magento\Catalog\Helper\Image::getUrl
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Helper/Image.php#L539
	vendor/magento/module-catalog/Helper/Image.php:539

9	Magento\Catalog\Ui\DataProvider\Product\Listing\Collector\Image::emulateImageCreating
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Ui/DataProvider/Product/Listing/Collector/Image.php#L155
	vendor/magento/module-catalog/Ui/DataProvider/Product/Listing/Collector/Image.php:155

10	call_user_func_array
	0

11	Magento\Framework\App\State::emulateAreaCode
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/App/State.php#L187
	vendor/magento/framework/App/State.php:187

12	Magento\Catalog\Ui\DataProvider\Product\Listing\Collector\Image::collect
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Ui/DataProvider/Product/Listing/Collector/Image.php#L115
	vendor/magento/module-catalog/Ui/DataProvider/Product/Listing/Collector/Image.php:115

13	Magento\Catalog\Ui\DataProvider\Product\ProductRenderCollectorComposite::collect
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Ui/DataProvider/Product/ProductRenderCollectorComposite.php#L36
	vendor/magento/module-catalog/Ui/DataProvider/Product/ProductRenderCollectorComposite.php:36

14	Magento\Catalog\Block\Ui\ProductViewCounter::getCurrentProductData
	https://github.com/magento/magento2/tree/2.4.6-p4/app/code/Magento/Catalog/Block/Ui/ProductViewCounter.php#L154
	vendor/magento/module-catalog/Block/Ui/ProductViewCounter.php:154

15	include
	var/view_preprocessed/pub/static/vendor/magento/module-catalog/view/frontend/templates/product/view/counter.phtml:5

16	Magento\Framework\View\TemplateEngine\Php::render
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/TemplateEngine/Php.php#L71
	vendor/magento/framework/View/TemplateEngine/Php.php:71

17	Magento\Framework\View\TemplateEngine\Php::___callParent
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L58
	vendor/magento/framework/Interception/Interceptor.php:58

18	Magento\Framework\View\TemplateEngine\Php::Magento\Framework\Interception\{closure}
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L138
	vendor/magento/framework/Interception/Interceptor.php:138

19	Magento\Framework\View\TemplateEngine\Php::___callPlugins
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L153
	vendor/magento/framework/Interception/Interceptor.php:153

20	Magento\Framework\View\TemplateEngine\Php::render
	generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php:23

21	Magento\Framework\View\Element\Template::fetchView
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Element/Template.php#L263
	vendor/magento/framework/View/Element/Template.php:263

22	Magento\Framework\View\Element\Template::_toHtml
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Element/Template.php#L293
	vendor/magento/framework/View/Element/Template.php:293

23	Magento\Framework\View\Element\AbstractBlock::Magento\Framework\View\Element\{closure}
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Element/AbstractBlock.php#L1128
	vendor/magento/framework/View/Element/AbstractBlock.php:1128

24	Magento\Framework\View\Element\AbstractBlock::_loadCache
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Element/AbstractBlock.php#L1132
	vendor/magento/framework/View/Element/AbstractBlock.php:1132

25	Magento\Framework\View\Element\AbstractBlock::toHtml
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Element/AbstractBlock.php#L676
	vendor/magento/framework/View/Element/AbstractBlock.php:676

26	Magento\Framework\View\Layout::_renderBlock
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L578
	vendor/magento/framework/View/Layout.php:578

27	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L555
	vendor/magento/framework/View/Layout.php:555

28	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

29	Magento\Framework\View\Layout::_renderContainer
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L606
	vendor/magento/framework/View/Layout.php:606

30	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L557
	vendor/magento/framework/View/Layout.php:557

31	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

32	Magento\Framework\View\Layout::_renderContainer
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L606
	vendor/magento/framework/View/Layout.php:606

33	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L557
	vendor/magento/framework/View/Layout.php:557

34	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

35	Magento\Framework\View\Layout::_renderContainer
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L606
	vendor/magento/framework/View/Layout.php:606

36	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L557
	vendor/magento/framework/View/Layout.php:557

37	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

38	Magento\Framework\View\Layout::_renderContainer
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L606
	vendor/magento/framework/View/Layout.php:606

39	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L557
	vendor/magento/framework/View/Layout.php:557

40	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

41	Magento\Framework\View\Layout::_renderContainer
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L606
	vendor/magento/framework/View/Layout.php:606

42	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L557
	vendor/magento/framework/View/Layout.php:557

43	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

44	Magento\Framework\View\Layout::_renderContainer
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L606
	vendor/magento/framework/View/Layout.php:606

45	Magento\Framework\View\Layout::renderNonCachedElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L557
	vendor/magento/framework/View/Layout.php:557

46	Magento\Framework\View\Layout::renderElement
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L510
	vendor/magento/framework/View/Layout.php:510

47	Magento\Framework\View\Layout::getOutput
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Layout.php#L975
	vendor/magento/framework/View/Layout.php:975

48	Magento\Framework\View\Layout::___callParent
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L58
	vendor/magento/framework/Interception/Interceptor.php:58

49	Magento\Framework\View\Layout::Magento\Framework\Interception\{closure}
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L138
	vendor/magento/framework/Interception/Interceptor.php:138

50	Magento\Framework\View\Layout::___callPlugins
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L153
	vendor/magento/framework/Interception/Interceptor.php:153

51	Magento\Framework\View\Layout::getOutput
	generated/code/Magento/Framework/View/Layout/Interceptor.php:41

52	Magento\Framework\View\Result\Page::render
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Result/Page.php#L260
	vendor/magento/framework/View/Result/Page.php:260

53	Magento\Framework\View\Result\Layout::renderResult
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/View/Result/Layout.php#L171
	vendor/magento/framework/View/Result/Layout.php:171

54	Magento\Framework\View\Result\Page::___callParent
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L58
	vendor/magento/framework/Interception/Interceptor.php:58

55	Magento\Framework\View\Result\Page::Magento\Framework\Interception\{closure}
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L138
	vendor/magento/framework/Interception/Interceptor.php:138

56	Magento\Framework\View\Result\Page::___callPlugins
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L153
	vendor/magento/framework/Interception/Interceptor.php:153

57	Magento\Framework\View\Result\Page::renderResult
	generated/code/Magento/Framework/View/Result/Page/Interceptor.php:41

58	Magento\Framework\App\Http::launch
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/App/Http.php#L120
	vendor/magento/framework/App/Http.php:120

59	Magento\Framework\App\Http::___callParent
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L58
	vendor/magento/framework/Interception/Interceptor.php:58

60	Magento\Framework\App\Http::Magento\Framework\Interception\{closure}
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L138
	vendor/magento/framework/Interception/Interceptor.php:138

61	Df\Framework\Plugin\App\Http::aroundLaunch
	https://github.com/mage2pro/core/tree/11.4.2/Framework/Plugin/App/Http.php#L10
	vendor/mage2pro/core/Framework/Plugin/App/Http.php:10

62	Magento\Framework\App\Http::Magento\Framework\Interception\{closure}
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L135
	vendor/magento/framework/Interception/Interceptor.php:135

63	Magento\Framework\App\Http::___callPlugins
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/Interception/Interceptor.php#L153
	vendor/magento/framework/Interception/Interceptor.php:153

64	Magento\Framework\App\Http::launch
	generated/code/Magento/Framework/App/Http/Interceptor.php:23

65	Magento\Framework\App\Bootstrap::run
	https://github.com/magento/magento2/tree/2.4.6-p4/lib/internal/Magento/Framework/App/Bootstrap.php#L264
	vendor/magento/framework/App/Bootstrap.php:264
```

**Comment by dmitrii-fediuk** [comment]
![2024-10-06--15-50-44](https://github.com/user-attachments/assets/4532c4e3-cbc0-438d-a29e-5f9a6e08c283)

**Comment by dmitrii-fediuk** [comment]
Related to:
- https://github.com/magento/magento2/pull/36679#issue-1513654627
- https://github.com/magento/magento2/pull/37319#issue-1649141031

**Comment by dmitrii-fediuk** [comment]
![02](https://github.com/user-attachments/assets/08d966ed-b104-4db7-848e-c1e523f1af47)

**Comment by dmitrii-fediuk** [comment]
>The changes requested in this PR are already merged in 2.4-develop branch by internal tickets **ACP2E-1632**.
https://github.com/magento/magento2/pull/37319#issuecomment-1495516812

**Comment by dmitrii-fediuk** [comment]
Fixed in Magento **≥ 2.4.7** by:
- https://github.com/magento/magento2/commit/edaa1583c2d361d62712e9102596050a9c13feac
- https://github.com/magento/magento2/commit/ecf36e9badc01ec520cb6551f78688d5414d7fce
- https://github.com/magento/magento2/commit/0da55f173115e940cb8e33dee2428618df9b4cbf
